### PR TITLE
Feature/cs4360 288 invoice model

### PR DESCRIPTION
**Pull request by jjohnson509**
working invoice service/model/controller

### CODE REVIEW

**Review by Newms226 — CHANGES_REQUESTED** [review]
Please comment on changes and let me know if you think its ready to merge :)

File: pb-crm-api/src/db/entities/payment.entity.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  @IsDefined()
+  @Length(2, 255)
+  method: string
+  
+  @OneToOne(() => AccountEntity, account => account.paymentInfo, {
+    cascade: ['update', 'insert'],
+    onUpdate: 'CASCADE',
+    onDelete: 'CASCADE'
+  })
+  account: AccountEntity;
```
[the remark this answers is not in the document]
Comment (jjohnson509) [reply pb-crm-api/src/db/entities/payment.entity.ts:29]: this is the inverse relation, and doesnt do validation. you cant really post payment info with an account. but can post an account with payment info, so the validation would come from the account entity.

File: pb-crm-api/src/contact/services/address.service.ts [hunk 2]
```diff
@@ -23,26 +24,36 @@ export class AddressService {
   }
 
   private async create(contact: ContactEntity, dto: AddressDTO, entityManager: EntityManager): Promise<AddressEntity> {
-    const newAddress: AddressEntity = entityManager.create<AddressEntity>(AddressEntity, { ...dto, contact });
+    const newAddress: AddressEntity = entityManager.create<AddressEntity>(AddressEntity, { ...dto });
```
[the remark this answers is not in the document]
Comment (jjohnson509) [reply pb-crm-api/src/contact/services/address.service.ts]: good catch, mistakenly left if out. added it back in.

**Comment by jjohnson509** [comment]
Everything should be good, I made changes, and left comments where appropriate. @Newms226

File: pb-crm-api/src/db/entities/payment.entity.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Newms226) [reply pb-crm-api/src/db/entities/payment.entity.ts:29]: 👍🏼

**Review by Newms226 — APPROVED** [review]
Changes look great, merging now

**Pull request merged**